lib/jekyll/utils.rb

```diff
@@ -297,5 +287,36 @@ def merged_file_read_opts(site, opts)
       merged
     end
 
+    private
+    def merge_values(target, overwrite)
+      target.merge!(overwrite) do |_key, old_val, new_val|
+        if new_val.nil?
+          old_val
+        elsif mergable?(old_val) && mergable?(new_val)
+          deep_merge_hashes(old_val, new_val)
+        else
+          new_val
+        end
+      end
+    end
+
+    private
+    def merge_default_proc(target, overwrite)
+      if default_proc?(target) && default_proc?(overwrite) && target.default_proc.nil?
+        target.default_proc = overwrite.default_proc
+      end
+    end
+
+    private
+    def default_proc?(object)
```

Comment:
> Do you suggest to change the if condition to check if both target and overwrite are Hash?

That is one solution, yes. I don't think `Drop`s support `default_proc`, so there is no need to try to merge a default proc for anything but a Hash.